### ISSUE TITLE
Feedback on page: /guides/v2.4/extension-dev-guide/events-and-observers.html

### ISSUE THREAD

**Issue by Lennonka** [magento/devdocs#7587]
# General issue

## Description

The documentation is missing a reference of events which come with Magento out of the box.

## Possible solutions

Include an official list and description of events would be most welcome.

## Additional information

These could be an inspiration:

- https://www.mageplaza.com/magento-2-module-development/magento-2-events.html#list-all-events-in-magento-2
- https://cyrillschumacher.com/magento-2.3-list-of-all-dispatched-events/

Even better if the official reference contained also a human description of when the events are emitted.

**Comment by goivvy** [comment]
@shrielenee  Will a new topic with all the events work? I can work on it

**Comment by shrielenee** [comment]
Hey @goivvy --- Yep! We would love if you would work on it. 🌟

**Comment by goivvy** [comment]
@magento I am working on this

**Comment by Lennonka** [comment]
Love you for working on this!

**Comment by goivvy** [comment]
Please see a screenshot - would this layout be informative enough? I'd like to use for the rest 300+ events, thought I'd ask first.
![Screen Shot 2020-11-20 at 5 04 19 PM](https://user-images.githubusercontent.com/1328492/99854451-d5403680-2b52-11eb-9c01-c993e959bb6e.png)

**Comment by goivvy** [comment]
@shrielenee Could you take a look, please?

**Comment by shrielenee** [comment]
Hey @goivvy Do you have an open or in-progress PR for this? It's easier to review that way. Calling in some reinforcements... paging @keharper 📲

**Comment by keharper** [comment]
This request is a job for automation. Any manually gathered list could be out of date the next patch release, and we can't search the code base every time we release a patch or add new functionality to the code base. 

@dshevtsov Does this look like something we can automate?

**Comment by dshevtsov** [comment]
I think so, but I will need time to explore available options.
cc @jeff-matthews

**Comment by goivvy** [comment]
Here is what I did:

1. Downloaded the latest Magento release
2. find vendor -type f -exec grep -iHn --color=always '\->dispatch(' {} \; >eventlist.txt
3. Went over eventlist.txt manually and store all event names in DB.

**Comment by goivvy** [comment]
@shrielenee @dshevtsov @keharper  Should I continue going over the events manually?

**Comment by Serializator** [comment]
It depends on what level of detail you want.

A simple overview of events can be done by using a regular expression. If you want some more detail then it'd quickly go to using some kind of AST to determine what is e.g. passed as data to the event.

An AST might be over-engineering the problem but at the same time feels like to only viable solution which can also be automated.

What do you think @goivvy?

**Comment by goivvy** [comment]
I'd root for a manual approach, that way it would be more accurate and I can add usage (when to listen) details which would be impossible to compose automatically.

**Comment by dshevtsov** [comment]
Unfortunately, manual approach would require manual reapplying for each new release.
We added this request to the list of automation requests in our backlog.

**Comment by goivvy** [comment]
That's 20 hours of my time wasted down the drain. Thank you.

**Comment by dshevtsov** [comment]
@goivvy, since we have encouraged you to work on this pull request and you have already done great work, it makes perfect sense to publish it and keep until we come up with the automation. We'll need to add corresponding notes referring to the documented components (code repositories such as magento2, inventory, etc) and versions of the code.

@shrielenee , @keharper what do you think?

**Comment by keharper** [comment]
I concur.

**Comment by goivvy** [comment]
OK.

**Comment by goivvy** [comment]
Will you keep my top 'thank you' backlink once you adopt an automated page?

**Comment by goivvy** [comment]
@dshevtsov @keharper @shrielenee Could you please answer my question before I move forward?

**Comment by jeff-matthews** [comment]
@goivvy, yes, we'll preserve the backlink in honor of your original coontribution.

**Comment by dobooth** [comment]
Hi @goivvy Did a PR ever come of this? It would be great to publish it.

**Comment by goivvy** [comment]
@dobooth I'm still working on it. 50+ events to go. Do you want me to push what's been already done and put the rest is coming up shortly notice at the bottom?

**Comment by goivvy** [comment]
@dobooth 
![Screen Shot 2021-03-25 at 5 02 06 PM](https://user-images.githubusercontent.com/1328492/112543253-dfd23d00-8d8b-11eb-9100-d23de350d547.png)

**Comment by goivvy** [comment]
@dobooth Could you comment on it?

**Comment by dobooth** [comment]
Sorry for the delay, @goivvy. If you still have it, we would love to see a PR with what you have so far.

**Comment by goivvy** [comment]
@dobooth https://github.com/magento/devdocs/pull/9098 The first 154 events.
